docs/guides/context.md

```diff
@@ -323,17 +323,22 @@ const countMachine = createMachine({
 
 ## TypeScript
 
-For proper type inference, add the context type as the first type parameter to `createMachine<TContext, ...>`:
+For proper type inference, add the context type to the schema property of the machine:
 
 ```ts
+import { t, createMachine } from 'xstate';
```

Comment:
I wonder if we should emphasize `t` helper here - or if that should only be something for people with aggressive linters.

Kinda fine either way for me, the only reason why I'm just a little bit hesitant about `t` as the default is that I'm worried about the quality of the IDE auto-imports for it. The identifier is so short (to improve part of the DX) that it feels like some other stuff might get often preferred. From my PoV using `{} as` is just as fine as this as it's super easy to write it and there are no gotchas to it (no more than with `t`). It feels "simpler" to me.

But as mentioned - I'm fine both ways, both will work anyway.

Reply:
`as` has also that small advantage that we don't have to care about adding `t` to the appropriate import when inserting the typegen meta to the file

Reply:
We discussed today (after you left this comment) that we'd only do linting, not auto-refactoring on the schema, so we're good - I think.

Reply:
We've been discussing refactors from generics to schema property but you still need to insert `as typeof import('').Typegen0` at the position of `tsTypes` property. So if we are recommending `t` then it will look somewhat weird that this will use a different pattern (`as` over `t`) or we'll have to potentially import `t` somehow.